client.go

```diff
@@ -1375,3 +1398,455 @@ func (c *ToznySDKV3) UnbrokerShare(ctx context.Context, authorizerClientID strin
 		RecordType:   recordType,
 	})
 }
+
+type SecretRequest struct {
+	SecretType  string
+	SecretName  string
+	SecretValue string
+	Description string
+	FileName    string
+}
+
+// CreateSecret makes a secret of the specified type and shares it with the client
+func (c *ToznySDKV3) CreateSecret(ctx context.Context, login TozIDLoginRequest, secret SecretRequest) (*pdsClient.Record, string, error) {
+	var createdSecret *pdsClient.Record
+	SECRET_TYPES := []string{"Client", "Credential", "File"}
+	// actually need to check the input for secret here too
+	secret.SecretName = strings.TrimSpace(secret.SecretName)
+	secret.SecretValue = strings.TrimSpace(secret.SecretValue)
+	if secret.SecretType == "" {
+		return createdSecret, "type cannot be empty", nil
```

Comment:
Use https://golang.org/pkg/errors/#example_New here, and you can combine the string and error return argument into one!